src/encoded/static/components/filegallery.js

```diff
@@ -388,6 +415,55 @@ FileTable.refTableColumns = {
 };
 
 
+const sortBioReps = (a, b) => {
+    // Sorting function for biological replicates of the given files.
+    let result; // Ends sorting loop once it has a value
+    let i = 0;
+    let repA = (a && a.biological_replicates && a.biological_replicates.length) ? a.biological_replicates[i] : undefined;
+    let repB = (b && b.biological_replicates && b.biological_replicates.length) ? b.biological_replicates[i] : undefined;
+
+    while (result === undefined) {
+        if (repA !== undefined && repB !== undefined) {
+            // Both biological replicates have a value
+            if (repA !== repB) {
+                // We got a real sorting result
+                result = repA - repB;
+            } else {
+                // They both have values, but they're equal; go to next
+                // biosample replicate array elements
+                i += 1;
+                repA = a.biological_replicates[i];
+                repB = b.biological_replicates[i];
+            }
+        } else if (repA !== undefined || repB !== undefined) {
+            // One and only one replicate empty; sort empty one after
+            result = repA ? 1 : -1;
+        } else {
+            // Both empty; sorting result same
+            result = 0;
+        }
+    }
+    return result;
+};
+
+const FileStatusLabel = (props) => {
+    const { file } = props;
+    const status = file.status;
+    const statusClass = globals.statusClass(status, 'status-indicator status-indicator--', true);
+
+    // Display simple string and optional title in badge
+    return (
+        <div key={status} className={statusClass}>
+            <i className="icon icon-circle status-indicator__icon" />
+            <div className="status-indicator__label">{status}</div>
+        </div>
+    );
+};
+
+FileStatusLabel.propTypes = {
+    file: PropTypes.object.isRequired, // File whose status we're displaying
+};
+
```

Comment:
Where is FileStatusLabel used? I don’t see it called from anywhere. eslint also doesn’t give an error for me, so that’s kind of odd.

Reply:
@forresttanaka I am still digging into this. FileStatusLabel was removed and replaced with a Status controller. I am digging into this.